SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -65,75 +64,127 @@ Plot {
 			backgroundColor = ~background;
 			this.gridLineSmoothing = ~gridLineSmoothing;
 			this.gridLinePattern = ~gridLinePattern !? {~gridLinePattern.as(FloatArray)};
-			labelX = ~labelX;
+			labelX = ~labelX; // axis label
 			labelY = ~labelY;
 		};
 		resolution = plotter.resolution;
+		prevBoundsSize = Size(0, 0);
 	}
 
 	bounds_ { |viewRect|
-		var gridRect;
-		var xtkLabels, ytkLabels, ytkLabelSize, xtkLabelSize, xLabelGridOffset, yLabelGridOffset;
-		var xLabelSize, yLabelSize, maxWidthWithLabel, maxHeightWithLabel;
-		var hshift, vshift, hinset, vinset, lmargin, rmargin, totalVSpace;
-		var minGridMargin = 4;
+		var gridRect, boundsSize;
+		var xTkLabelWd, xAxLabelHt, yAxLabelWd, yTkLabelHt, bottomLabelHt;
```

Comment:
Given the size of this method, perhaps you have an idea of how to factor the functionality into either a helper method or a dedicated class that calculates such things?

Reply:
See comment above about this now being factored out to `Plot:-prCalcLabelSpace`.

Reply:
Ah yes, that makes it cleaner. I suppose one can't factor out the three return values into separate methods …

One general thing: if you want the compiler to optimise boolean expressions, don't forget the parentheses.
Instead of `wdAllowsYLabels and: htAllowsYLabels` it should be `wdAllowsYLabels and: { htAllowsYLabels }`.

Then, one very general question (sorry!): why do we need the complicated calculation? Couldn't we just resize the view according to what is needed?

Reply:
> why do we need the complicated calculation? Couldn't we just resize the view according to what is needed?

It stems from the fact that the _axis_ labels are independent from _tick_ labels, and further independent between _axes_. The labels on each axis are hidden if the corresponding dimension of the view is small enough that the plot:label ratio goes below a threshold (otherwise the plot becomes useless because it's all labels and tiny data). 

So the calculation gets really tricky depending which labels are active (subject to user settings and plot size)...

> I suppose one can't factor out the three return values into separate methods …

... and these interdependencies make factoring them out quite a challenge (though future work might be done here).

Reply:
> One general thing: if you want the compiler to optimise boolean expressions, don't forget the parentheses.
Instead of wdAllowsYLabels and: htAllowsYLabels it should be wdAllowsYLabels and: { htAllowsYLabels }

Thanks, I didn't realize this optimization also applied to evaluation of single boolean values 👍